### ISSUE TITLE
Garbled encoding of ID3 tags

### ISSUE THREAD

**Issue by vasyl-shumskyi** [Borewit/music-metadata#193]
Just curious if your awesome module correctly transforms [garbled encoding](http://en.wikipedia.org/wiki/Mojibake) of ID3 tags to readable tags (i.e UTF-8)

The track which is not processed correctly is the following: 

[Download from Dropbox](https://www.dropbox.com/s/e9x9o0mpyefk7ze/01-Ja_pidu_v_daleki_gori.mp3?dl=1)

I believe such transformation is done by checking encoding bytes first

> $00 – ISO-8859-1 (LATIN-1, Identical to ASCII for values smaller than 0x80).
> $01 – UCS-2 (UTF-16 encoded Unicode with BOM), in ID3v2.2 and ID3v2.3.
> $02 – UTF-16BE encoded Unicode without BOM, in ID3v2.4.
> $03 – UTF-8 encoded Unicode, in ID3v2.4.

Thanks!

**Comment by Borewit** [comment]
Hi Vasyl.

As far as I know it should _correctly_ decode the ID3 tags, using the specified encoding. 
If not, I will look into it.

What do you mean with correctly transforms garbled encoding?
If tags are set to a Mojibake value, one way or another way, (like was done here #164), you will get back Mojibake.

I did not have time to study your file yet.

**Comment by vasyl-shumskyi** [comment]
> What do you mean with correctly transforms garbled encoding?

I meant correct decoding of regular non-Mojibake tags. 

Sorry for the mix-up. Mojibake style was linked for the reference of garbled encoding look only.

**Comment by Borewit** [comment]
This is the output of music-metadata, as displayed by [audio-tag-analyzer](https://audio-tag-analyzer.netlify.com/):
![music-metadata issue 193 audio-tag-analyzer](https://user-images.githubusercontent.com/23255260/52880216-a917aa80-3161-11e9-8f2d-11949288ba4d.png)

Which is consistent with other good implementation:

![music-metadata issue foobar2000 193](https://user-images.githubusercontent.com/23255260/52879969-fa736a00-3160-11e9-9539-774e19a766f9.png)
![music-metadata issue 193 mp3tags](https://user-images.githubusercontent.com/23255260/52879974-fe06f100-3160-11e9-90d0-55570825bbee.png)

The [Mojibake](https://en.wikipedia.org/wiki/Mojibake) is s result of [GIGO](https://en.wikipedia.org/wiki/Garbage_in,_garbage_out).

In other words, looks like that audio file is not correctly encoded, or the encoder settings are wrong.

**Comment by Borewit** [comment]
The original file also had an ID3v1 header, that is simple not able to store `Плач Єремії` / `Я піду в далекі гори`.

I updated the tag values to demonstrate it can be done correctly:

[01-Ja_pidu_v_daleki_gori [fixed].zip](https://github.com/Borewit/music-metadata/files/2871757/01-Ja_pidu_v_daleki_gori.fixed.zip)

I used [Picard](https://picard.musicbrainz.org/) to identify the track, and updated the metadata fields accordingly. Saved as an ID3v2.3, like the the original bit without the ID3v1.

**Comment by vasyl-shumskyi** [comment]
Hey @Borewit Thanks for your time on debugging this! 

I had an assumption that this should be caused by ID3v1... But never figured out how to get ID3 version info correctly. 

I tried with python mutagen and MusicBrainz Picard 
The latest Picard version showed me ID3v2.3 so I gave up and created issue here :)

```
$ mutagen-inspect 01-Ja_pidu_v_daleki_gori.mp3

-- 01-Ja_pidu_v_daleki_gori.mp3
- MPEG 1 layer 3, 256000 bps, 44100 Hz, 271.47 seconds (audio/mp3)
TIT2=ß ï³äó â äàëåê³ ãîðè
TLEN=271000
TPE1=Ïëà÷ ªðåìi¿
```

<img width="1447" alt="screen shot 2019-02-23 at 19 59 06" src="https://user-images.githubusercontent.com/2331802/53289947-1fda2680-37a6-11e9-9765-6863108fb66c.png">
